src/main/java/com/sandvoxel/generitech/common/tileentities/machines/TileEntityPulverizer.java

```diff
@@ -201,151 +201,160 @@ public boolean canBeRotated() {
-
-                    if (worldObj.isRemote)
-                        return;
+            ticksRemaining = 200;
+            machineActive = true;
+            this.markForUpdate();
+            this.markDirty();
+        }
 
-                    ItemStack processItem = inventory.getStackInSlot(1);
-                    if (processItem == null) {
-                        return;
-                    }
+        if (inventory.getStackInSlot(1) != null && ticksRemaining <= 0) {
 
-                    List<Crushable> outputs = PulverizerRegistry.getOutputs(processItem);
+            ticksRemaining = 0;
 
-                    if (outputs.isEmpty())
-                        return;
+            if (worldObj.isRemote) {
+                return;
+            }
 
-                    for (int i = this.crushIndex; i < outputs.size(); i++) {
-                        this.crushIndex = i;
-                        Crushable crushable = outputs.get(this.crushIndex);
+            ItemStack processItem = inventory.getStackInSlot(1);
+            if (processItem == null) {
+                return;
+            }
 
-                        ItemStack outItem = crushable.output.copy();
-                        float itemChance = crushable.chance;
-                        boolean itemFortune = crushable.luckMultiplier == 1.0f;
+            List<Crushable> outputs = PulverizerRegistry.getOutputs(processItem);
 
-                        if (crushRNG == -1) crushRNG = rnd.nextFloat();
+            if (outputs.isEmpty())
+                return;
 
-                        if (itemFortune)
-                            itemChance = itemChance + fortuneMultiplier;
+            for (int i = this.crushIndex; i < outputs.size(); i++) {
+                this.crushIndex = i;
+                Crushable crushable = outputs.get(this.crushIndex);
 
-                        outItem.stackSize = (int) Math.round(Math.floor(itemChance) + crushRNG * itemChance % 1);
-                        if (outItem.stackSize == 0) outItem.stackSize = 1;
+                ItemStack outItem = crushable.output.copy();
+                float itemChance = crushable.chance;
+                boolean itemFortune = crushable.luckMultiplier == 1.0f;
 
+                if (crushRNG == -1) crushRNG = rnd.nextFloat();
 
-                        // Simulate placing into output slot...
-                        if (InventoryHelper.addItemStackToInventory(outItem, inventory, 2, 3, true) != null) {
-                            this.pulverizerPaused = true;
-                            return;
-                        }
+                if (itemFortune)
+                    itemChance = itemChance + fortuneMultiplier;
 
-                        if (pulverizerPaused)
-                            pulverizerPaused = !pulverizerPaused;
+                outItem.stackSize = (int) Math.round(Math.floor(itemChance) + crushRNG * itemChance % 1);
+                if (outItem.stackSize == 0) outItem.stackSize = 1;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS00122)